crypto/bio/bss_conn.c

```diff
@@ -208,8 +208,18 @@ static int conn_state(BIO *b, BIO_CONNECT *c)
                 ERR_raise(ERR_LIB_BIO, BIO_R_NBIO_CONNECT_ERROR);
                 ret = 0;
                 goto exit_loop;
-            } else
+            } else {
                 c->state = BIO_CONN_S_OK;
+# ifndef OPENSSL_NO_KTLS
+                /*
+                 * The new socket is created successfully regardless of ktls_enable.
+                 * ktls_enable doesn't change any functionality of the socket, except
+                 * changing the setsockopt to enable the processing of ktls_start.
+                 * Thus, it is not a problem to call it for non-TLS sockets.
+                 */
+                ktls_enable(b->num);
```

Comment:
Hmm, in my tests this call retuns sometimes ENOTCONN.
So it's an improvement, but does not 100% work for nonblocking connects,
and that's what doit_localhost does.
The man page says
```
EINPROGRESS
              The socket is nonblocking and the connection cannot be completed
              immediately.   (UNIX domain sockets failed with EAGAIN instead.)
              It is possible to select(2) or poll(2) for completion by select‐
              ing the socket for writing.  After select(2) indicates writabil‐
              ity, use getsockopt(2) to read  the  SO_ERROR  option  at  level
              SOL_SOCKET to determine whether connect() completed successfully
              (SO_ERROR is zero) or unsuccessfully (SO_ERROR  is  one  of  the
              usual  error  codes  listed  here, explaining the reason for the
              failure).
```
we are not doing the select here, probably that's the problem,
I don't see how the bss_conn.c object can signal a SSL_ERROR_WANT_WRITE...